### PR TITLE
Components

### PR DESCRIPTION
**Pull request by shradhaj06**
Created the sidebar component.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.app {
+    display: flex;
+  }
+  
+  .main-content {
+    margin-left: 60px; /* Same width as the sidebar */
+    padding: 20px;
+    width: calc(100% - 60px);
```
Comment (Jha2508) [inline src/App.css:8]: Why this? Why arent we using margins of 60px

File: src/Components/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const [activeIcon, setActiveIcon] = useState('home');
+
+  return (
+    <div className="sidebar-container">
+      <div className="sidebar">
+        <Link to="/home">
+          <img
+            src={homeIcon}
+            alt="Home"
+            className={activeIcon === 'home' ? 'active' : ''}
```
Comment (Jha2508) [inline src/Components/Sidebar/Sidebar.jsx]: Move these to constants

Comment (Jha2508) [reply src/Components/Sidebar/Sidebar.jsx]: Home and active

File: src/Components/Sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            onClick={() => setActiveIcon('home')}
+          />
+        </Link>
+        <Link to="/ticket">
+          <img
+            src={ticketIcon}
+            alt="Ticket"
+            className={activeIcon === 'ticket' ? 'active' : ''}
```
Comment (Jha2508) [inline src/Components/Sidebar/Sidebar.jsx]: Move to constants

File: src/index.js [hunk 4]
```diff
@@ -1,17 +1,12 @@
...
 import App from './App';
-import reportWebVitals from './reportWebVitals';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
-);
+    <BrowserRouter>
```
Comment (Jha2508) [inline src/index.js:21]: Why is there a need to remove the strictMode? Document it before removing in the code as comments

File: .vscode/tasks.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+    "tasks": [
+        {
+            "type": "cppbuild",
+            "label": "C/C++: g++.exe build active file",
+            "command": "C:\\msys64\\mingw64\\bin\\g++.exe",
+            "args": [
```
Comment (Jha2508) [inline .vscode/tasks.json]: Why is this committed?

**Review by Jha2508 — CHANGES_REQUESTED** [review]

File: src/App.css (hunk 1 above)
Comment (shradhaj06) [reply src/App.css:8]: Using a margin of 60px on all sides of .main-content would push it away from surrounding elements, potentially causing layout issues and misalignment. By specifically setting the left margin to 60px, the main content area is shifted away from the sidebar while keeping the rest of its layout unaffected.
The calc(100% - 60px) width calculation ensures that the main content area takes up the remaining width after subtracting the width of the sidebar, making it responsive to changes in screen size or layout adjustments.

Comment (shradhaj06) [reply src/App.css:8]: I will be working on disabling mobile view of the application after this.

File: src/Components/Sidebar/Sidebar.jsx (hunk 3 above)
Comment (shradhaj06) [reply src/Components/Sidebar/Sidebar.jsx]: done

File: .vscode/tasks.json (hunk 5 above)
Comment (shradhaj06) [reply .vscode/tasks.json]: has been removed

File: src/index.js (hunk 4 above)
Comment (shradhaj06) [reply src/index.js:21]: documentation added

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**